[Flaky Test]: TestComponentBuildHashInDiagnostics – The process cannot access the file because it is being used by another process.

### Failing test case

TestComponentBuildHashInDiagnostics

### Error message

The process cannot access the file because it is being used by another process.

### Build

https://buildkite.com/elastic/elastic-agent/builds/8425#018ef032-8659-4f38-86e1-c33e2fd48587

### OS

Windows

### Stacktrace and notes

```markdown
package_version_test.go:161: creating directory "C:\\Users\\windows\\AppData\\Local\\Temp\\TestComponentBuildHashInDiagnostics4261895062\\003\\logs\\elastic-agent-8.14.0-SNAPSHOT-44319e\\http"
    package_version_test.go:161: unzipping file "C:\\Users\\windows\\AppData\\Local\\Temp\\TestComponentBuildHashInDiagnostics4261895062\\003\\logs\\elastic-agent-8.14.0-SNAPSHOT-44319e\\system"
    package_version_test.go:161: creating directory "C:\\Users\\windows\\AppData\\Local\\Temp\\TestComponentBuildHashInDiagnostics4261895062\\003\\logs\\elastic-agent-8.14.0-SNAPSHOT-44319e\\system"
    fixture_install.go:214: [test TestComponentBuildHashInDiagnostics] Inside fixture cleanup function
    fixture.go:656: >> running binary with: [C:\Program Files\Elastic\Agent\elastic-agent.exe uninstall --force]
    testing.go:1225: TempDir RemoveAll cleanup: remove C:\Users\windows\AppData\Local\Temp\TestComponentBuildHashInDiagnostics4261895062\003\state.yaml: The process cannot access the file because it is being used by another process.
--- FAIL: TestComponentBuildHashInDiagnostics (64.67s)
```

--- a/testing/integration/package_version_test.go
+++ b/testing/integration/package_version_test.go
@@ -160,8 +160,13 @@ func TestComponentBuildHashInDiagnostics(t *testing.T) {
 	diag := t.TempDir()
 	extractZipArchive(t, diagZip, diag)
 
-	stateYAML, err := os.Open(filepath.Join(diag, "state.yaml"))
+	stateFilePath := filepath.Join(diag, "state.yaml")
+	stateYAML, err := os.Open(stateFilePath)
 	require.NoError(t, err, "could not open diagnostics state.yaml")
+	defer func(stateYAML *os.File) {
+		err := stateYAML.Close()
+		assert.NoErrorf(t, err, "error closing %q", stateFilePath)
+	}(stateYAML)
 
 	state := struct {
 		Components []struct {

